packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -3918,9 +3992,11 @@ export function attach(
         reconcilingParent = stashedParent;
         previouslyReconciledSibling = stashedPrevious;
         remainingReconcilingChildren = stashedRemaining;
-        reconcilingParentSuspenseNode = stashedSuspenseParent;
-        previouslyReconciledSiblingSuspenseNode = stashedSuspensePrevious;
-        remainingReconcilingChildrenSuspenseNodes = stashedSuspenseRemaining;
+        if (fiberInstance.suspenseNode !== null) {
+          reconcilingParentSuspenseNode = stashedSuspenseParent;
+          previouslyReconciledSiblingSuspenseNode = stashedSuspensePrevious;
+          remainingReconcilingChildrenSuspenseNodes = stashedSuspenseRemaining;
+        }
```

Comment:
I think `unmountFiberRecursively` in  https://github.com/sebmarkbage/react/blob/35b14817c6ee519cd3ac4ddaf7c205cd9f2b964c/packages/react-devtools-shared/src/backend/fiber/renderer.js#L3166-L3168 is also missing the conditional restore.